pxd.c

```diff
@@ -1040,8 +1042,10 @@ static blk_status_t pxd_queue_rq(struct blk_mq_hw_ctx *hctx,
 	struct request *rq = bd->rq;
 	struct pxd_device *pxd_dev = rq->q->queuedata;
 	struct fuse_req *req = blk_mq_rq_to_pdu(rq);
+	struct fuse_conn *fc = &pxd_dev->ctx->fc;
 
-	if (BLK_RQ_IS_PASSTHROUGH(rq))
+	if (BLK_RQ_IS_PASSTHROUGH(rq) ||
+		(!fc->allow_disconnected && !fc->connected))
```

Comment:
early return based on allow_disconnected and connected state. This ensures no allocation and immediate freeing of requests happen later.